client/landing/gutenboarding/components/header/index.tsx

```diff
@@ -74,6 +75,16 @@ const Header: FunctionComponent< Props > = ( {
 		</span>
 	);
 
+	const domainElement = (
+		<span
+			className={ classnames( 'gutenboarding__header-domain-picker-button-domain', {
+				placeholder: ! currentDomain,
+			} ) }
+		>
+			{ currentDomain ? currentDomain.domain_name : 'example.wordpress.com' }
+		</span>
+	);
```

Comment:
I think this should live inside the DomainPickerButton component for better encapsulation. Also, instead of `children` and `disabled` it could take just `currentDomain` prop.

Reply:
Yeah, was wondering about where to delineate components and concerns while working on this :thinking: 

I'm not totally sure we want `currentDomain` as a prop for `DomainPicker`, since the latter is supposed to be rather agnostic about the user's current choice.

Reply:
Oops, I might've conflated `DomainPicker` and `DomainPickerButton` a bit in my previous comment :grimacing: